zerver/migrations/0699_set_namedusergroup_rendered_description.py

```diff
@@ -0,0 +1,56 @@
+# Generated by Django 5.0.10 on 2025-01-16 14:46
+
+from django.db import migrations, transaction
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+from django.db.models import Max, Min
+
+
+def render_all_user_group_descriptions(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    # FIXME: Application code should not be imported from migrations.
+    from zerver.actions.user_groups import render_group_description
+
+    NamedUserGroup = apps.get_model("zerver", "NamedUserGroup")
+    BATCH_SIZE = 1000
+    max_id = NamedUserGroup.objects.exclude(description="").aggregate(Max("id"))["id__max"]
+    if max_id is None:
+        # Do nothing if there are no user groups on the server.
+        return
+
+    lower_bound = NamedUserGroup.objects.exclude(description="").aggregate(Min("id"))["id__min"]
+
+    while lower_bound <= max_id:
+        upper_bound = lower_bound + BATCH_SIZE - 1
+        print(f"Processing batch {lower_bound} to {upper_bound} for NamedUserGroup")
+
+        with transaction.atomic():
+            all_named_user_groups = NamedUserGroup.objects.exclude(
+                id__gte=lower_bound, id__lt=upper_bound, description=""
+            )
+            for user_group in all_named_user_groups:
+                user_group.rendered_description = render_group_description(
+                    user_group.description, user_group.realm
+                )
```

Comment:
I don't think it's likely safe to run the Markdown processor inside a transaction.